docs/providers/aws/guide/building.md

```diff
@@ -20,6 +20,8 @@ In Serverless Framework V.4, [esbuild](https://github.com/evanw/esbuild) is incl
 
 By default, if your AWS Lambda handler is using Typescript files directly, the Framework will build your code automagically upon deploy, without a plugin. No configuration is necessary by default.
 
+**Note:** Even though you are writing ESM code, you should not set "type": "module" in your package.json file, as your code will be transpiled to CommonJS.
```

Comment:
This is not a correct assumption, there is a difference to writing ESM based typescript. If we are seeing issues here then it is something wrong with our esbuild support not recognizing and building ESM correctly. See [esbuild docs](https://esbuild.github.io/api/#format), the build output format can be `esm`